### ISSUE TITLE
Add "surrender round" vote for attacking team in stopwatch mode

### ISSUE THREAD

**Issue by BystryPL** [etlegacy/etlegacy#1688]
On some maps after certain time it's obvious that the attacking team won't do the objective in time (for example on supply, when allies can't take the flag on 1st stage of the map). This can prolong the whole match by couple of minutes that are spent on useless shooting. I suggest, if possible, adding voting option for the attacking team to surrender the round (not the whole match) if they are sure that they won't make the objective in time.

**Comment by Aciz** [comment]
There is already surrender vote (and a ref command to surrender too) but the vote is disabled on configs by default. The cvar to allow it is `vote_allow_surrender`. It surrenders the current round (not the whole match).

**Comment by rmarquis** [comment]
We might need to ensure the clock is correctly set in case of surrender (artificially set to the defined time limit).

**Comment by Aciz** [comment]
It correctly sets the clock to fullhold if attackers surrender, and if defenders surrender, it will set the clock to whatever the time was when the vote passed.

**Comment by rmarquis** [comment]
Perfect then. Closing the ticket.

**Comment by rmarquis** [comment]
>  it will set the clock to whatever the time was when the vote passed.

Oops, I misread again. The clock should be set as if the team never achieved the obj until the timelimit finishes. Otherwise, it would be obviously buggy in second round.

**Comment by Aciz** [comment]
I think you misread again :D this is the case if DEFENDERS surrender. If ATTACKERS surrender, the clock will be set to full round time.

**Comment by rmarquis** [comment]
I just realized that... gonna stop reading Github for the day. My bad.

Here's a question though: is there any advantage of the defending team to surrender? Basically they'd shot themselves in the foot, as they would shorten their clock in round 2 (if surrendering in round 1). Shouldn't we prevent surrender by the defending team entirely in SW mode?

**Comment by Aciz** [comment]
Yeah I can't think of any reason really for defenders to surrender, unless they want to make it a challenge for themselves :P So yeah makes no sense really, but I'd say it's also not very critical, you still need to consciously make the decision to surrender the round, it's not like it can happen by accident (unless a defending team member has ref access).

**Comment by rmarquis** [comment]
Attempt to justify the ticket reopening and not appear like a complete fool: SUCCESSFUL :D

Anyway, I think it doesn't hurt to prevent the defending team to surrender. There is no reason it should happen, so let's not allow them to do it.

**Comment by mittermichal** [comment]
Imagine you defend sw_goldrush_te against much better team:
Tanks gets repaired 30 seconds into map, everyone in your team is dead and waiting to get spawned back. Meanwhile tank is slowly moving down hill. Your team respawns, but attacking team already got field ops with artillery pointed at spawn exit. Your whole team dies again. The attacking engineer who is moving tank went to build command post, so the tank even stopped. Your team spawns, but everyone gets killed again because whole attacking team except the engineer is camping each spawn exit. Engineer builds command post and forgets he is supposed to move tank and joins the spawn killing fest :D You know where this is going :) Oh and by the way it takes minute to escort the truck with gold, that gives the enemy field ops another 2 chances of making some fragmovie airstrike :D
Yes, I've been in this situation many times.

In this case I think it makes sense for defenders to surrender the whole map (2nd round isn't played)

**Comment by Aciz** [comment]
In that case you might as well just vote for the next map and count the score being 2-0, I don't think having a "surrender whole map" option is really needed.

**Comment by mittermichal** [comment]
in case you want to have some kind of scoring system implemented on the server, It would be better to tell the server that it win for attacking team, so the score changes

**Comment by Aciz** [comment]
An in-game scoring system would be a nice addition indeed, in which case what you described above does make sense indeed.
